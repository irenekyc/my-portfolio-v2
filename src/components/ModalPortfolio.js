import React from 'react'
import {useSelector, useDispatch} from 'react-redux'
import {closePortfolio} from '../actions/modal'


const ModalPortfolio =()=>{
    const projectdata = useSelector(state=> state.data)
    const dispatch = useDispatch()
  
    return(

        projectdata!==null && projectdata.projectDetails !==null &&(
            <div className="portfolio-modal">
            <div className="portfolio-details">
                <div id="close-portfolio"> <span onClick={()=>dispatch(closePortfolio())}>X </span></div>
                <div className="portfolio-details-title">
                    <h1>
                        { projectdata.projectDetails.title}
                    </h1>
                </div>
                <div className="portfolio-details-image">
                    <img src={`images/${projectdata.projectDetails.image}`} alt={ projectdata.projectDetails.title} />
                </div>
                <div className="portfolio-details-descriptions">
                    <p>{ projectdata.projectDetails.excerpt}</p>
                    <p>{ projectdata.projectDetails.fullDescription}</p>
                </div>
                <div className="portfolio-details-tools">
                    <h4> Tools & Libraries </h4>
                    <p>{ projectdata.projectDetails.tools}</p>
                </div>
                <div className="links">
                <a href={ projectdata.projectDetails.projectUrl} rel="noopener noreferrer" target="_blank">  <i className="fas fa-globe"></i> - Live Site</a>
                <a href={ projectdata.projectDetails.github} rel="noopener noreferrer" target="_blank"><i className="fab fa-github"></i> - Code</a>
            </div>
            </div>

        </div>
        )
       
    )
}

export default ModalPortfolio