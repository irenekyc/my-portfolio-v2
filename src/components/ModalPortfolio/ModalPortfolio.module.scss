.ikModal {
  &__dialog {
    color: #eee;
  }
  &__content {
    background: #303841 !important;

    color: #eee;
    max-width: 700px;
    margin-right: auto;
    margin-left: auto;
  }
  &__header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    padding: 1rem 1.5rem;
  }
  &__title {
    font-size: 1.5rem;
  }
  &__image {
    height: 200px;
    overflow: hidden;
    margin-bottom: 1rem;

    img {
      width: auto;
      height: 100%;
    }
  }
  &__descriptions {
    padding: 0.5rem 0;
    line-height: 1.8;
  }
  &__tools {
    padding: 0.5rem 0;
    h4 {
      color: #b7985b;
      font-size: 1.2rem;
      margin-bottom: 0.5rem;
    }
  }
  &__button {
    padding: 0.3rem 1rem;
    border: 1px solid #eee;
    text-decoration: none;
    color: #eee;

    &:hover {
      color: #eee;
    }
  }
}
