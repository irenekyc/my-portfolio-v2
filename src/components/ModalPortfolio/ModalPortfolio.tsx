import React, { FunctionComponent } from "react";
import IKModal from "../../layout/Modal";
import { Project } from "../../typings/Project";
import Modal from "react-bootstrap/Modal";
import styles from "./ModalPortfolio.module.scss";

interface ModalPortfolioProps {
  show: boolean;
  clickClose: () => void;
  projectDetails: Project | null;
}

const ModalPortfolio: FunctionComponent<ModalPortfolioProps> = ({
  show,
  clickClose,
  projectDetails,
}: ModalPortfolioProps) => {
  if (!projectDetails) return <></>;
  return (
    <IKModal
      show={show}
      onHide={clickClose}
      dialogClassName={styles.ikModal__dialog}
      contentClassName={styles.ikModal__content}
      size="xl"
    >
      <Modal.Header
        closeButton
        bsPrefix={styles.ikModal__header}
        closeVariant="white"
      >
        <h2 className={styles.ikModal__title}>{projectDetails.title}</h2>{" "}
      </Modal.Header>
      <Modal.Body>
        <div className={styles.ikModal__image}>
          <img
            src={`images/${projectDetails.image}`}
            alt={projectDetails.title}
          />
        </div>
        <div className={styles.ikModal__descriptions}>
          <p>{projectDetails.excerpt}</p>
          <p>{projectDetails.fullDescription}</p>
        </div>
        <div className={styles.ikModal__tools}>
          <h4> Tools & Libraries </h4>

          <p>{projectDetails.tools.join(", ")}</p>
        </div>
        <div className="links">
          <a
            className={styles.ikModal__button}
            href={projectDetails.projectUrl}
            rel="noopener noreferrer"
            target="_blank"
          >
            {" "}
            <i className="fas fa-globe"></i> - Live Site
          </a>
          <a
            className={styles.ikModal__button}
            href={projectDetails.github}
            rel="noopener noreferrer"
            target="_blank"
          >
            <i className="fab fa-github"></i> - Code
          </a>
        </div>
      </Modal.Body>
    </IKModal>
  );
};

export default ModalPortfolio;
