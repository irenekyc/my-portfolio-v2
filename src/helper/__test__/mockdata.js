export const mockProject = {
  title: "title",
  category: "category",
  tools: [],
  mainTag: [],
  image: "image.png",
  excerpt: "some excerpt",
  fullDescription: "some description",
  projectUrl: undefined,
  github: "https://github.com",
};

export const mockProjectListLong = [
  {
    title: "title1",
    category: "category",
    tools: ["Next"],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title2",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title3",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title4",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title5",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title6",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title7",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title8",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },

  {
    title: "title9",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title10",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title11",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
];
