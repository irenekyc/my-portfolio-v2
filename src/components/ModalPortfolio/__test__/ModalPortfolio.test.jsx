import React from "react";
import ModalPortfolio from "../index";
import { render, screen } from "@testing-library/react";
import { mockProject } from "../../../helper/__test__/mockdata";

const mockClose = jest.fn();

it("Render ModalPortfolio Section - should not show by default when project details are not passed", () => {
  render(
    <ModalPortfolio show={false} clickClose={mockClose} projectDetails={null} />
  );

  const modalPortfolio = screen.queryByTestId("modal-portfolio");
  expect(modalPortfolio).toBeNull();
});

it("Render ModalPortfolio Section - should not show when show is false", () => {
  render(
    <ModalPortfolio
      show={false}
      clickClose={mockClose}
      projectDetails={mockProject}
    />
  );

  const modalPortfolio = screen.queryByTestId("modal-portfolio");
  expect(modalPortfolio).toBeNull();
});

it("Render ModalPortfolio Section - should show when show is true, and correctly render details", () => {
  render(
    <ModalPortfolio
      show={true}
      clickClose={mockClose}
      projectDetails={mockProject}
    />
  );

  const modalPortfolio = screen.queryByTestId("modal-portfolio");
  expect(modalPortfolio).toBeInTheDocument();
  expect(modalPortfolio).toMatchSnapshot();
  const modalPortfolioTitle = screen.getByTestId("modal-portfolio-title");
  expect(modalPortfolioTitle).toHaveTextContent(mockProject.title);
  const modalPortfolioImage = screen.getByTestId("modal-portfolio-image");
  expect(modalPortfolioImage).toHaveAttribute(
    "src",
    `images/${mockProject.image}`
  );
  const modalPortfolioExcerpt = screen.getByTestId("modal-portfolio-excerpt");
  expect(modalPortfolioExcerpt).toHaveTextContent(mockProject.excerpt);
  const modalPortfolioDescription = screen.getByTestId(
    "modal-portfolio-description"
  );
  expect(modalPortfolioDescription).toHaveTextContent(
    mockProject.fullDescription
  );
  const modalPortfolioProjectUrl = screen.queryByTestId(
    "modal-portfolio-project-url"
  );
  expect(modalPortfolioProjectUrl).toBeNull();
  const modalPrtoflioGithubUrl = screen.queryByTestId("modal-portfolio-github");
  expect(modalPrtoflioGithubUrl).toHaveAttribute("href", mockProject.github);
});
