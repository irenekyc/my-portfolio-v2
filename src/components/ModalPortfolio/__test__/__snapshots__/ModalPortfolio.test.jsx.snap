// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ModalPortfolio Section - should show when show is true, and correctly render details 1`] = `
<div
  class="modal-dialog ikModal__dialog modal-xl"
  data-testid="modal-portfolio"
>
  <div
    class="modal-content ikModal__content"
  >
    <div
      class="ikModal__header"
    >
      <h2
        class="ikModal__title"
        data-testid="modal-portfolio-title"
      >
        title
      </h2>
       
      <button
        aria-label="Close"
        class="btn-close btn-close-white"
        type="button"
      />
    </div>
    <div
      class="modal-body"
    >
      <div
        class="ikModal__image"
      >
        <img
          alt="title"
          data-testid="modal-portfolio-image"
          src="images/image.png"
        />
      </div>
      <div
        class="ikModal__descriptions"
      >
        <p
          data-testid="modal-portfolio-excerpt"
        >
          some excerpt
        </p>
        <p
          data-testid="modal-portfolio-description"
        >
          some description
        </p>
      </div>
      <div
        class="ikModal__tools"
      >
        <h4>
           Tools & Libraries 
        </h4>
        <p />
      </div>
      <div
        class="links"
      >
        <a
          class="ikModal__button"
          data-testid="modal-portfolio-github"
          href="https://github.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="fab fa-github"
          />
           - Code
        </a>
      </div>
    </div>
  </div>
</div>
`;
